src/components/shop-image.js

```diff
@@ -0,0 +1,104 @@
+/**
+ * @license
+ * Copyright (c) 2018 The Polymer Project Authors. All rights reserved.
+ * This code may only be used under the BSD style license found at http://polymer.github.io/LICENSE.txt
+ * The complete set of authors may be found at http://polymer.github.io/AUTHORS.txt
+ * The complete set of contributors may be found at http://polymer.github.io/CONTRIBUTORS.txt
+ * Code distributed by Google as part of the polymer project is also
+ * subject to an additional IP rights grant found at http://polymer.github.io/PATENTS.txt
+ */
+
+import { LitElement, html } from '../../node_modules/@polymer/lit-element/lit-element.js';
+
+class ShopImage extends LitElement {
+  render({ alt, placeholder, _src, _style }) {
+    return html`
+    <style>
+
+      :host {
+        display: block;
+        position: relative;
+      }
+
+      #placeholder {
+        position: absolute;
+        top: 0;
+        bottom: 0;
+        left: 0;
+        right: 0;
+        overflow: hidden;
+        background-size: cover;
+        background-position: center;
+      }
+
+      img {
+        max-width: 100%;
+        max-height: 100%;
+        margin: 0 auto;
+        opacity: 0;
+        transition: 0.5s opacity;
+        
+        position: absolute;
+        top: 0;
+        bottom: 0;
+        left: -9999px;
+        right: -9999px;
+        max-width: none;
+      }
+
+    </style>
+
+    <div id="placeholder" style="${`background-image: url('${placeholder}')`}">
+      <img src="${_src}"
+          alt="${alt}"
+          style="${_style}"
+          on-load="${() => this._onImgLoad()}"
+          on-error="${() => this._onImgError()}">
+    </div>
+`;
+  }
+
+  static get is() { return 'shop-image'; }
+
+  static get properties() { return {
+
+    alt: String,
+
+    src: String,
+
+    placeholder: String,
+
+    _src: String,
+
+    _style: String
+
+  }}
+
+  _propertiesChanged(props, changed, oldProps) {
+    if (changed && 'src' in changed) {
+      this._srcChanged();
+    }
+    super._propertiesChanged(props, changed, oldProps);
+  }
+
+  async _srcChanged() {
+    this._src = '';
+    this._style = 'opacity: 0';
+    await this.nextRendered;
+    if (this.src) {
+      this._src = this.src;
```

Comment:
`CIRC (changes in response to changes) considered harmful` ... need to avoid this. Is it necessary since to blank src since you're setting opacity? (is that for sizing?)

Reply:
Tried a few things, but seems like the image needs to have a blank src for at least one frame to achieve the effect when changing from one src to another (e.g. going from one category to another).